rfd/0170-package-distribution-v3.md

```diff
@@ -0,0 +1,426 @@
+  needs on the same scale as us. I am not confident that they have the size or
+  speed needed to support us.
+* To company's About page, their tools are "Windows first, not as an
+  afterthought." They go on to say, "Instead, we prefer Windows, and always
+  start development, testing, and documentation in Windows." While this may be a
+  selling point for some of their customers, we are a Linux and macOS driven
+  shop. I suspect that the ProGet will not be able to support us as well as if
+  we were one of their target customers. This is corroborated in [one of their
+  docs
+  pages](https://docs.inedo.com/docs/proget-feeds-other-types#:~:text=For%20example%2C%20when,use%20these%20languages).
+
+This is not a viable solution for us.
+
+##### Nexus Repository
+
+Sonatype Nexus Repository could be a great fit for us. It can be self hosted,
+supports both APT and YUM repos without customer-facing changes, and can be
+managed via GitOps (with a third-party Terraform provider). However,
+customer-facing download requests would need to flow through a service that we
+are responsible for keeping online and available, failing this requirement.
+
+##### Artifactory
+
+JFrog Artifactory meets all of our requirements, and is the *only* solution on
+the market today that does so. In addition to meeting our requirements, we would
+see several other benefits with this solution:
+
+* Artifactory has support for several other repository types that we currently
+  have. By (eventually) using Artifactory for these, we could get rid of a lot
+  of tooling and infrastructure. Here's a list of repos types that we could
+  shift to this solution:
+  * Helm
+    * Currently requires custom tooling for our release pipelines
+  * Terraform (Nexus Repository requires a third-party plugin)
+    * Currently requires extensive custom tooling for our release pipelines, as
+      well as tooling for disaster recovery
+  * Release server/raw binary artifacts
+    * Currently requires a large amount of tooling in a monolithic service that
+      struggles to support our current needs
+* There is support for pull-through caches and/or mirroring for all major
+  toolchains that we use in our product today. By using Artifactory (eventually)
+  as a pull-through cache, we could scan all dependencies for vulnerabilities,
+  malicious code, and changes in licensing prior to download. This would not
+  only increase stability of our release pipelines, but also improve the
+  security of our developer endpoints. here's a list of package types that are
+  supported here:
+  * OS packages (debs, rpms)
+    * We don't pin these at all in our release pipelines, meaning we currently
+      blindly trust the upstream sources
+  * Go modules
+    * We pin these currently but do not actively scan these for new/unreported
+      vulnerabilities
+  * Rust creates
+    * We pin these currently but do not actively scan these for new/unreported
+      vulnerabilities
```

Comment:
We have GitHub's Dependabot security alerts and regular `govulncheck` runs, so I don't know if it's fair to say we don't actively scan for vulnerabilities.

Reply:
These don't check for unreported vulnerabilities, correct? My understanding (I can ask JFrog for clarification here if desired) is that JFrog's tool will look for issues in modules that have not been discovered yet/had CVE filed for yet.